packages/aws-sqs/src/predicate/assert-message-interface.mts

```diff
@@ -0,0 +1,48 @@
+import type { MessageInterface } from "../definition/interface/message.interface.mjs";
+import type { MessageAttributeInterface } from "../definition/interface/message-attribute.interface.mjs";
+
+import { assertMessageAttributeInterface } from "./assert-message-attribute-interface.mjs";
+import { assertArray, assertRecord, assertString, assertStructuredData } from "@vitruvius-labs/ts-predicate/type-assertion";
+
+function assertMessageInterface(value: unknown): asserts value is MessageInterface
+{
+	assertStructuredData<MessageInterface>(value, {
+		Attributes: {
+			test: (scopedValue: unknown): asserts scopedValue is Record<string, unknown> =>
```

Comment:
Suggestion to normalize names in assertor: reuse the name of the property (here: `attributes`)

Invalid assertion: `asserts scopedValue is Record<string, unknown>`
Should be: `asserts scopedValue is Array<MessageAttributeInterface>`